admin/src/components/login.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-console */
+import React, { Component } from 'react';
+
+import logo from '../assets/SmallLogo.png';
+
+class LogIn extends Component {
+  state = {
+    email: '',
+    password: '',
+    error: '',
+    message: '',
+  }
+
+  componentDidMount() {
+    const { token } = this.props;
+    const storedToken = window.localStorage.getItem('token');
```

Comment:
This component is accessing the data that the same component is writing in the local storage, but App also manages it by deleting it when it finds it convenient.

This breaks the separation of concerns. Let's centralize the management of the `localStorage` in one place so it can be maintained easily. i.e.: a component providing a context; a redux middleware implementing `localStorage` side-effects.